code/processes/chainedtp.q

```diff
@@ -174,7 +174,7 @@ end:{[d]
   /- roll over the log you need a new log for next days data 
   .ctp.refreshtp[d+1];
   /- push endofday messages to subscribers
-  (neg union/[w[;;0]])@\:(`.u.end;d)
+  (neg union/[.stpps.allsubhandles[]])@\:(`.u.end;d)
```

Comment:
Could / should this be both lists of handles? 

i.e. w[;;0],.stpps.allsubhandles[]

Reply:
Idea would work except .u.w isn't defined straight off the bat in sctp. Sctp would need a .u.init[] or similar to have the .u.w table defined off the bat

Reply:
But in checking this, causes a type error

Reply:
Could you make it 

neg union[@[value;(`.stpps.allsubhandles;`);()]; @[{union/[(value x)[;;0]};`.u.w;()]]

Basically we need to maintain backward compatibility here as well as making sure it works for new. There are probably a number of different ways that people could have this deployed - so if we change to this I think it catches most cases - whether they are using new or old subscription logic (or some mixture) I think it should work. It shouldn't need to have .u.w defined as that should be error trapped